salt/utils/versions.py

```diff
@@ -18,18 +18,39 @@
 import sys
 import warnings
 
-from setuptools._distutils.version import LooseVersion as _LooseVersion
-from setuptools._distutils.version import StrictVersion as _StrictVersion
+log = logging.getLogger(__name__)
+
+try:
+    from setuptools.distutils.version import LooseVersion as _LooseVersion
+except ImportError:
+    log.debug(
+        "unable to import LooseVersion from setuptools.distutils.version, will attempt setuptools._distutils.version"
+    )
+    try:
+        from setuptools._distutils.version import LooseVersion as _LooseVersion
+    except:
+        log.debug("unable to import LooseVersion from setuptools._distutils.version")
+        raise ImportError()
+
+try:
+    from setuptools.distutils.version import StrictVersion as _StrictVersion
+except ImportError:
+    log.debug(
+        "unable to import StrictVersion from setuptools.distutils.version, will attempt setuptools._distutils.version"
+    )
+    try:
+        from setuptools._distutils.version import StrictVersion as _StrictVersion
+    except:
+        log.debug("unable to import StrictVersion from setuptools._distutils.version")
+        raise ImportError()
+
```

Comment:
```suggestion
try:
    from setuptools.distutils.version import LooseVersion as _LooseVersion
    from setuptools.distutils.version import StrictVersion as _StrictVersion
except ImportError:
    from setuptools._distutils.version import LooseVersion as _LooseVersion
    from setuptools._distutils.version import StrictVersion as _StrictVersion
```
Let's just try/catch both alternatives.

Reply:
And because of the ssh errors, a third fallback into importing from distutils, since the looseversion package is not being installed.

Reply:
Wanted to isolate the import so there was individual log messages. I can combine them and just have the one generic log message, as there should be a log message imho.

Reply:
Don't log import errors, unless there's some usefulness to that.
Also don't raise `ImportError()`, raise the original exception.

Reply:
Updated to similar with log.debug messages